### PR TITLE
[DOC] Add config for ssl ciphers

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Documented the proeprties that now allow you to run external listeners on specific ciphers for a TLS version.

Updated descriptions of the following properties

- Kafka brokers: spec.kafka.config //including sample YAML
- Zookeeper: spec.zookeeper.config
- Kafka Connect: spec.config
- Kafka Connect S2I: spec.config
- Kafka Mirror Maker (1.0): spec.consumer.config
- Kafka Mirror Maker (1.0): spec.producer.config
- Kafka Mirror Maker (2.0): spec.clusters.config 
- Kafka Bridge: kafkaBridge.spec.producer.config
- Kafka Bridge: kafkaBridge.spec.consumer.config

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by PaulRMellor** [comment]
MM and MM2 treated slightly differently.
The configuration is described in the schema, as we're in the process of moving the descriptions of MM properties to the schema reference.

_Might not want both links from the MirrorMaker 1.0 example config_

**Comment by scholzj** [comment]
I pushed the autogenerated files and rebased to get rid of a conflict - not hopefully the builds pass. You will need to pull first before doing ny further changes @PaulRMellor

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Use `ssl` configuration options to run external listeners on a specific _cipher suite_ for a TLS version.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc:1]: > Use the three allowed `ssl` configuration options to run external listeners on a specific _cipher suite_ for a TLS version.
> A _cipher suite_ combines algorithms for secure connection and data transfer.

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Use the three allowed `ssl` configuration options to run external listeners on a specific _cipher suite_ for a TLS version.
```
Comment (ppatierno) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc:1]: is it "on a specific cipher suite" the right one? or should it be "with a specific cipher suite"? Same comment for all the other files.

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    config:
+      ssl.cipher.suites: "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384" <1>
+      ssl.enabled.protocols: "TLSv1.2" <2>
+      ssl.protocol: "TLSv1.2" <3>
+----
+<1> The cipher suite for TLS using a combination of `ECDHE` key exchange mechanism, `RSA` authentication algorithm,
+`AES` bulk encyption algorithm and `SHA384` MAC algorithm.
+<2> The SSl protocol `TLSv1.2` is enabled.
+<3> Specifies the `TLSv1.2` protocol to generate the SSL context.
+Allowed values are `TLSv1.2` and `TLSv1.2`.
```
Comment (ppatierno) [inline documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc:18]: `TLSv1.2` and `TLSv1.2` ? Same question for all the other files.

Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc:18]: Allowed values are `TLSv1`, `TLSv1.1` and `TLSv1.2`.

File: documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc:1]: _with_ is better

**Comment by scholzj** [comment]
Looks like ti worked now 👍

**Pull request merged**